### DISCUSSION TITLE
Unable to Flash Xavier NX with initrd-flash

### DISCUSSION THREAD

**Discussion by rishabnayak** [OE4T/meta-tegra#1764] [Q&A]
I've built `core-image-sato-dev` using meta-tegra:scarthgap-l4t-r35.x and poky:scarthgap for the `jetson-xavier-nx-devkit-emmc`, following instructions to setup udev rules and polkit config, and set `TNSPEC_BOOTDEV:jetson-xavier-nx-devkit-emmc = "nvme0n1p1"` in my local.conf. I've also turned off automount.

When I attempt flashing with initrd-flash, I receive this error

```
Starting at 2024-11-14T12:15:23-05:00
Machine:       jetson-xavier-nx-devkit-emmc
Rootfs device: nvme0n1p1
Found Jetson device in recovery mode at USB 1-3
== Step 1: Signing binaries at 2024-11-14T12:15:23-05:00 ==
== Step 2: Boot Jetson via RCM at 2024-11-14T12:15:33-05:00 ==
Found Jetson device in recovery mode at USB 1-3
== Step 3: Sending flash sequence commands at 2024-11-14T12:15:41-05:00 ==
Waiting for USB storage device flashpkg from 4c3ffa05......[/dev/sdc]
Device size in blocks: 262144
Unmounted /dev/sdc.
== Step 4: Writing boot partitions to internal storage device at 2024-11-14T12:15:59-05:00 ==
Waiting for USB storage device mmcblk0 from 4c3ffa05...[/dev/sdc]
Creating partitions
ERR: could not initialize GPT on /dev/sdc
ERR: write failure to internal storage at 2024-11-14T12:16:02-05:00
```

I believe I'm missing the device-tree config to enable OTG mode. How would I do that?

**Comment by dwalkes** [comment]
This might be related to https://github.com/OE4T/meta-tegra/pull/1394 - I'm wondering if something like this needs to be ported to scarthgap.

**Reply by rishabnayak** [reply]
I came across this PR when going down the device tree config rabbit hole yesterday. Happy to help out.
